src/docker/shared.rs

```diff
@@ -498,29 +494,65 @@ pub(crate) fn custom_image_build(
     let pre_build = config.pre_build(target)?;
 
     if let Some(pre_build) = pre_build {
-        if !pre_build.is_empty() {
-            let custom = Dockerfile::Custom {
-                content: format!(
-                    r#"
-    FROM {image}
-    ARG CROSS_DEB_ARCH=
-    ARG CROSS_CMD
-    RUN eval "${{CROSS_CMD}}""#
-                ),
-            };
-            custom
-                .build(
-                    config,
-                    metadata,
-                    engine,
-                    &host_root,
-                    Some(("CROSS_CMD", pre_build.join("\n"))),
-                    target,
-                    msg_info,
-                )
-                .wrap_err("when pre-building")
-                .with_note(|| format!("CROSS_CMD={}", pre_build.join("\n")))?;
-            image = custom.image_name(target, metadata)?;
+        match pre_build {
+            super::custom::PreBuild::Single(pre_build_script)
+                if !pre_build_script.contains('\n')
+                    && host_root.join(&pre_build_script).is_file() =>
+            {
+                let custom = Dockerfile::Custom {
+                    content: format!(
+                        r#"
+            FROM {image}
+            ARG CROSS_DEB_ARCH=
+            ARG CROSS_SCRIPT
+            COPY $CROSS_SCRIPT /script
+            RUN chmod +x /script
+            RUN ./script "#
```

Comment:
I'm not sure this is the correct way to do this, should maybe keep the filename.

I think using chmod is good though, even if it's not strictly needed

Reply:
It will probably help with me having to avoid pushing commits on Windows or commit, then change permissions on Linux and recommit, so I like the change.